compiler/qsc_lowerer/src/lib.rs

```diff
@@ -878,10 +878,57 @@ impl Lowerer {
             name_span: field.name_span,
         }
     }
-}
 
-fn lower_generics(generics: &[qsc_hir::ty::GenericParam]) -> Vec<qsc_fir::ty::GenericParam> {
-    generics.iter().map(lower_generic_param).collect()
+    fn lower_generics(
```

Comment:
This had to become a method, because a type parameter can now contain class constraint parameters which could be concrete types. Those concrete types need to be lowered. `.lower_ty()` is a method, so we therefore have to make this a method to have access to `self`.